internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -73,14 +82,36 @@ func (hr *HypershiftAgentServiceConfigReconciler) Reconcile(origCtx context.Cont
 	}
 
 	// Creating spoke client using specified kubeconfig secret reference
-	// TODO: use spoke client to create CRDs/CRs
-	_, err := hr.createSpokeClient(ctx, instance.Spec.KubeconfigSecretRef.Name, instance.Namespace)
+	spokeClient, err := hr.createSpokeClient(ctx, instance.Spec.KubeconfigSecretRef.Name, instance.Namespace)
 	if err != nil {
-		log.WithError(err).Error("Failed to create client using remote kubeconfig", req.NamespacedName)
+		log.WithError(err).Error("Failed to create client using specified kubeconfig", req.NamespacedName)
+		return ctrl.Result{}, err
+	}
+
+	// Ensure agent-install CRDs exist on spoke cluster and clean stale ones
+	if err := hr.syncSpokeAgentInstallCRDs(ctx, spokeClient); err != nil {
+		log.WithError(err).Error(fmt.Sprintf("Failed to sync agent-install CRDs on spoke cluster in namespace '%s'", instance.Namespace))
+		conditionsv1.SetStatusConditionNoHeartbeat(&instance.Status.Conditions, conditionsv1.Condition{
+			Type:    aiv1beta1.ConditionReconcileCompleted,
+			Status:  corev1.ConditionFalse,
+			Reason:  aiv1beta1.ReasonSpokeClusterCRDsSyncFailure,
+			Message: "Failed to sync agent-install CRDs on spoke cluster.",
```

Comment:
consider adding the the `err` here as well, otherwise users will not know if it's something that they can fix or not. 
for example if it's internal issue then no, but if there is no connection to the other cluster then maybe it's something that they can check at least.